dataprep/clean/clean_headers.py

```diff
@@ -0,0 +1,273 @@
+    remove_accents: bool = True,
+    report: bool = True,
+    progress: bool = True,
+) -> pd.DataFrame:
+    """
+    Function to clean headers / column names.
+
+    Parameters
+    ----------
+    df
+        Dataframe from which column names are to be cleaned.
+
+    case
+        The desired case style of the column name.
+            - 'snake': 'column_name'
+            - 'kebab': 'column-name'
+            - 'camel': 'columnName'
+            - 'pascal': 'ColumnName'
+            - 'const': 'COLUMN_NAME'
+            - 'sentence': 'Column name'
+            - 'title': 'Column Name'
+            - 'lower': 'column name'
+            - 'upper': 'COLUMN NAME'
+
+        (default: 'snake')
+
+    replace
+        Values to replace in the column names.
+            - {"old_value": "new_value"}
+
+        (default: None)
+
+    remove_accents
+        If True, strip accents from the column names.
+
+        (default: True)
+
+    report
+        If True, output the summary report. Otherwise, no report is outputted.
+
+        (default: True)
+
+    progress
+        If True, enable the progress bar.
+
+        (default: True)
+
+    Examples
+    --------
+    Clean column names by converting the names to camel case style, removing accents,
+    and correcting a mispelling.
+
+    >>> df = pd.DataFrame({"FirstNom":["Philip", "Turanga"], "lastName": ["Fry", "Leela"],
+                           "Téléphone":["555-234-5678", "(604) 111-2335"]})
+    >>> clean_headers(df, case='camel', replace={"Nom": "Name"})
+    Headers Cleaning Report:
+        2 values cleaned (66.67%)
+    Result contains 3 (100.0%) values in the correct format and 0 null values (0.0%)
+      firstName lastName       telephone
+    0    Philip      Fry    555-234-5678
+    1   Turanga    Leela  (604) 111-2335
+    """
+    # pylint: disable=too-many-arguments
+
+    case_styles = {
+        "snake",
+        "kebab",
+        "camel",
+        "pascal",
+        "const",
+        "sentence",
+        "title",
+        "lower",
+        "upper",
+    }
+
+    if case not in case_styles:
+        raise ValueError(
+            f"case {case} is invalid, it needs to be one of {', '.join(c for c in case_styles)}"
+        )
+
+    # Convert to dask
+    df = to_dask(df)
+
+    # Store original column names for creating cleaning report
+    orig_columns = df.columns
+
+    if replace:
+        df = df.rename(columns=lambda col: _replace_values(col, replace))
+
+    if remove_accents:
+        df = df.rename(columns=_remove_accents)
+
+    df = df.rename(columns=lambda col: _convert_case(col, case))
+
+    df.columns = _rename_duplicates(df.columns, case)
+
+    stats = {}
+    # Count the number of changed column names
+    stats["cleaned"] = len(df.columns.astype(str).difference(orig_columns.astype(str)))
+    # Count null column values
+    stats["null"] = len([col for col in list(df.columns) if col in NULL_VALUES])
+
+    with ProgressBar(minimum=1, disable=not progress):
```

Comment:
I think you can remove `dask.compute(df)` here, and also the progress bar parameter from `clean_headers`.